arangod/Aql/SharedQueryState.h

```diff
@@ -101,9 +101,33 @@ class SharedQueryState final
   /// execute a task in parallel if capacity is there
   template<typename F>
   bool asyncExecuteAndWakeup(F&& cb) {
+    // The atomic _numTasks counts the number of ongoing tasks asynchronous
+    // tasks. We need this such that we can wait for them to finish when
+    // the query is shut down. Note that this is *not* necessary for
```

Comment:
> We need this such that we can wait for them to finish when the query is shut down.

While that's true, I'd say the *main* reason for numTasks is to know whether we are allowed to spawn more asynchronous tasks, and we shouldn't keep quiet about this, lest we confuse the next reader for a different reason.

Reply:
Will amend the comment.